.github/workflows/docker-build-scan.yaml

```diff
@@ -0,0 +1,111 @@
+name: Docker Build and Scan Image
+
+on:
+  push:
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+-*'
+
+env:
+  PLATFORMS: linux/amd64,linux/arm64
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    environment: vault-init
+    permissions:
+      contents: read
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Configure Docker
+        id: meta
+        uses: ./.github/actions/configure-docker
+        with:
+          image_name: ${{ vars.IMAGE_NAME }}
+          registry_username: ${{ secrets.DOCKERHUB_USERNAME }}
+          registry_token: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: ${{ vars.CONTEXT }}
+          tags: ${{ steps.meta.outputs.image_tags }}
+          labels: ${{ steps.meta.outputs.image_labels }}
+          platforms: ${{ env.PLATFORMS }}
+          outputs: type=oci,dest=vault-init-image.tar
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+      - name: Upload image artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: vault-init-image
+          path: vault-init-image.tar
+
+  scan:
+    needs: build
+    runs-on: ubuntu-latest
+    environment: vault-init
+    permissions:
+      contents: read
+      actions: read
+      security-events: write
+
+    steps:
+      - name: Download image artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: vault-init-image
+          path: .
+
+      - name: Unarchive tarball
+        run: |
+          mkdir vault-init-image
+          tar -xvf vault-init-image.tar -C vault-init-image
+
+      - name: Scan Docker image for vulnerabilities
+        id: scan
+        uses: aquasecurity/trivy-action@0.16.1
+        with:
+          input: /github/workspace/vault-init-image
+          ignore-unfixed: true
+          severity: 'CRITICAL,HIGH'
+          exit-code: '1'
+          format: 'table'
```

Comment:
Let's capture all of them, but fail the build and alert for high and critical only

Reply:
If we want use SARIF/Code scanning reports, we can't fail our pipeline during scan action because it will prevent results from uploading. I'm thinking on two options:
1. (Lame) Make second scan right before pushing to Docker Hub.
2. Build and scan at branch push (restrict to script location/Dockerfile changes), build and push at tag push (will be restricted by ruleset if there high or higher errors or first build).

What do you think?

Reply:
Is there an option to create a some sort of an artifact or a variable that would contain a boolean value, based on which a push job is triggered?

Reply:
Found a workaround for how to run the upload step even if 'scan' found some vulnerabilities, but if SARIF isn't limited by our rule (we can upload a full file with 'limit-severities-for-sarif' parameter) it will always fail if some vulnerabilities are found.

So in that case we need to limit SARIF output to our HIGH and CRITICAL, until trivy-action change something.